### PR TITLE
Black Hat Programming 

### PR DESCRIPTION
**Pull request by lokori**

### CODE REVIEW

File: _posts/2018-05-02-Black-Hat-Programming.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+---
+
+
+## Enter Black Hat Programming
+
+Forget automated tests. Never mind Facade, Decorator or other OOP design patterns. Static type systems? Quickly, toss them out of the way. It's time to exploit some security flaws and no amount of so called "professional software development" is going to help with that. Welcome to the world of Black Hat Programming!
+
+![Black hat](/img/blackhat/blackhat.jpg)
+
+In lack of a better term, Black Hat Programming is the art of writing tools and scripts for penetration testing and expoiting vulnerabilities. Just don't go poking around without permission and become a Black Hat hunted by the police. But why don't we need all those "best practices" that are very useful normally in professional software development? The "black hat programming" discussed in this post typically deals with a simpler world:
```
Comment (tatut) [inline _posts/2018-05-02-Black-Hat-Programming.md]: I think the idiom is "For lack of a better term"

File: _posts/2018-05-02-Black-Hat-Programming.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+![King Python](/img/blackhat/python-book.jpg)
+
+## Let's hack!
+
+Okay, enough talk, let's write some Black Hat Python then, shall we? Remember, these are quickly written scripts that are good enough to get the job done and serve as templates for the next time something similar is needed. This is the craft of quickly making tools when a need arises, not a craft of making reliable distributed systems that are modular and maintainable.
+
+
+### Scanning ports behind a squid proxy
+
+If a [Squid HTTP proxy](http://www.squid-cache.org/) is not properly configured, it can  expose ports and services behind it that are not directly visible to outside world. This might give us more insight into the machine and othe systems connected to it. And this could even be abused to [smuggle SSH protocol over Squid proxy](https://daniel.haxx.se/docs/sshproxy.html). The poor proxy gets confused and thinks it's HTTP traffic.
```
Comment (tatut) [inline _posts/2018-05-02-Black-Hat-Programming.md]: "othe" missing r

File: _posts/2018-05-02-Black-Hat-Programming.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+* After the program works, it becomes pretty static. The vuln is what it is and the requirements do not change.
+* The end user is a hacker who knows about security, technical stuff and programming. The programmers are 
+  the first end users of their scripts.
+ 
+Because of these characteristics, there is no need to do sprints, UX design, paper prototypes, automated tests.
+You can kiss your pair programming goodbye. And what kind of person starts arguing about the coding style 
+and architectural patterns when the whole script is about 50 lines of code? If it works and the code
+is understandable, it's a success! O brave new world, that has such programmers in it!
+
+## Choose productivity, Choose Pwnage. Choose Python.
```
Comment (riikkanen) [inline _posts/2018-05-02-Black-Hat-Programming.md]: Sekavaa kirjainkokoa... pitäisikö productivykin olla isolla P:llä?

File: _posts/2018-05-02-Black-Hat-Programming.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+The field of security is wide, dealing often with bit operations, opcodes, crypto and hash algoritms,
+databases and network protocols. Therefore the ideal programming language needs to handle everything easily and 
+have a wide variety of libraries ready to do the heavy lifting. Java, C# and many others are qualified in this sense.
+
+But Java, C# and some other "professional" programming languages get left behind when getting things done becomes paramount. A 
+hacker doesn't want to spend time designing types and classes and wondering what **erasure** or **generics** means. It's
+about the speed of development - just hack together the idea, test until it stops bleeding.. and profit! [Cowboy coding](https://en.wikipedia.org/wiki/Cowboy_coding) is a natural fit.
```
Comment (riikkanen) [inline _posts/2018-05-02-Black-Hat-Programming.md]: .. -> ...

File: _posts/2018-05-02-Black-Hat-Programming.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+It is no wonder that Python is the absolute favorite of pen testers and hackers all over the world. Perl might become a distant second and lately there has been some interest in Go. And there is Ruby as  the widely used [Metasploit](https://www.metasploit.com/) is Ruby-based. But Python is the high-king of this realm and a very well liked king he is. Long may he reign!
```
Comment (riikkanen) [inline _posts/2018-05-02-Black-Hat-Programming.md]: 'as the', two spaces in between

File: _posts/2018-05-02-Black-Hat-Programming.md (hunk 2 above)
Comment (riikkanen) [inline _posts/2018-05-02-Black-Hat-Programming.md]: 'can expose', two spaces in between

**Pull request merged**